dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/calculate-fixed-fee-adjustment.test.ts

```diff
@@ -0,0 +1,56 @@
+import { when } from 'jest-when';
+import { FacilityUtilisationDataEntityMockBuilder, FeeRecordEntityMockBuilder, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { calculateFixedFeeAdjustment } from './calculate-fixed-fee-adjustment';
+import { aReportPeriod } from '../../../../../../test-helpers/test-data';
+import { calculateFixedFee } from './calculate-fixed-fee';
+
+jest.mock('./calculate-fixed-fee');
+
+describe('calculateFixedFeeAdjustment', () => {
+  beforeEach(() => {
+    jest.mocked(calculateFixedFee).mockRejectedValue(new Error('Some error'));
+  });
+
+  it('throws an error if the fee record facility id does not match the facility utilisation data id', async () => {
+    // Arrange
+    const feeRecord = FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withFacilityId('11111111').build();
+    const facilityUtilisationData = FacilityUtilisationDataEntityMockBuilder.forId('22222222').build();
+
+    // Act / Assert
+    await expect(calculateFixedFeeAdjustment(feeRecord, facilityUtilisationData, aReportPeriod())).rejects.toThrow(
+      new Error('Fee record facility id does not match the facility utilisation id'),
+    );
+  });
+
+  it('returns the difference between the current fixed fee and the facility utilisation data fixed fee rounded to 2 decimal places', async () => {
```

Comment:
I think we either should go down the path of 'current' and 'previous' or down the path of 'fee record fixed fee' and 'facility utilisation fixed fee' and my preference would be 'current' and 'previous'. So e.g. rename 'facility utilisation data fixed fee' in this test name to 'previous fixed fee' and then also in the test could rename:
- `feeRecordFacilityUtilisation` -> `currentFacilityUtilisation`
- `facilityUtilisationDataUtilisation` -> `previousFacilityUtilisation`
- `feeRecordFixedFee` -> `currentFixedFee`
- ...